shipitscript/shipitscript/script.py

```diff
@@ -61,15 +62,15 @@ def create_new_release_action(context):
         # TODO quit early, mark task as green though
         pass
     log.info('Determining most recent shippable revision')
-    shippable_revision = ship_actions.get_shippable_revision(
-        repo, last_shipped_revision
+    should_build = ship_actions.should_ship_revision(
+        repo, last_shipped_revision, current_revision,
     )
-    if not shippable_revision:
+    if not should_build:
         # TODO quit early, mark task as green though
         pass
     log.info('create a new release')
     ship_actions.create_new_release(
-        product, repo, channel, next_version, shippable_revision, phase, shipit_config
+        product, repo, channel, next_version, current_revision, phase, shipit_config
```

Comment:
so I'm still not sure this will work. We want to tell `create_new_release` what revision to ship from. This just takes the most tip revision every time. I get that if we have a commit in the pushlog of commits that is "important" we want to trigger a new release, but should we always take the tip rev? What if there are no builds on that rev? e.g. `NPOTB DONTBUILD`?

Reply:
It's now taking the `shippable_revision` or it's No-op and task goes green.